hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoDeleteAllowanceHandler.java

```diff
@@ -88,15 +89,24 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         // Every owner whose allowances are being removed should sign (or the payer, if there is no owner)
         for (final var allowance : op.nftAllowancesOrElse(emptyList())) {
             if (allowance.hasOwner()) {
-                context.requireKeyOrThrow(allowance.ownerOrThrow(), INVALID_ALLOWANCE_OWNER_ID);
+                final var store = context.createStore(ReadableAccountStore.class);
+                final var ownerId = allowance.ownerOrThrow();
+                final var owner = store.getAccountById(ownerId);
+                final var approvedForAll = owner.approveForAllNftAllowancesOrElse(emptyList()).stream()
+                        .anyMatch(approveForAll -> approveForAll.tokenId().equals(allowance.tokenId())
+                                && approveForAll.spenderId().equals(context.payer()));
+                if (approvedForAll) {
+                    context.requireKeyOrThrow(context.payerKey(), INVALID_DELEGATING_SPENDER);
```

Comment:
⚠️ This isn't necessary, and would actually break this call for a sender contract with a cryptographic admin key. I.e. we only need,
```
if (!context.payer().equals(ownerId) && !approvedForAll) {
         context.requireKeyOrThrow(ownerId, INVALID_ALLOWANCE_OWNER_ID);
}
```

Reply:
Done, thanks!